subscription integration test flakiness

seen in pr https://github.com/dapr/dapr/pull/9685
```

=== Failed
=== FAIL: tests/integration Test_Integration/daprd/hotreload/operator/informer/subscriptions/run (45.24s)
    integration.go:97: running test case
    subscriptions.go:191: 
        	Error Trace:	D:/a/dapr/dapr/tests/integration/framework/process/daprd/daprd.go:493
        	            				D:/a/dapr/dapr/tests/integration/framework/process/daprd/daprd.go:412
        	            				D:/a/dapr/dapr/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go:192
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/runtime/asm_amd64.s:1771
        	Error:      	Received unexpected error:
        	            	Get "http://127.0.0.1:6397/v1.0/metadata": context deadline exceeded
    subscriptions.go:191: 
        	Error Trace:	D:/a/dapr/dapr/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go:192
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/runtime/asm_amd64.s:1771
        	Error:      	"[]" should have 1 item(s), but has 0
    subscriptions.go:191: 
        	Error Trace:	D:/a/dapr/dapr/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go:191
        	            				D:/a/dapr/dapr/tests/integration/integration.go:98
        	Error:      	Condition never satisfied
        	Test:       	Test_Integration/daprd/hotreload/operator/informer/subscriptions/run
    daprd.go:282: 
        	Error Trace:	D:/a/dapr/dapr/tests/integration/framework/process/daprd/daprd.go:282
        	            				D:/a/dapr/dapr/tests/integration/framework/process/daprd/daprd.go:289
        	            				D:/a/dapr/dapr/tests/integration/framework/process/grpc/subscriber/subscriber.go:173
        	            				D:/a/dapr/dapr/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go:194
        	            				D:/a/dapr/dapr/tests/integration/integration.go:98
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/runtime/asm_amd64.s:1771
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	Test_Integration/daprd/hotreload/operator/informer/subscriptions/run
        --- FAIL: Test_Integration/daprd/hotreload/operator/informer/subscriptions/run (45.24s)

=== FAIL: tests/integration Test_Integration/daprd/hotreload/operator/informer/subscriptions (45.98s)
    subscriptions.go:75: environment variable DAPR_INTEGRATION_CRD_DIRECTORY not set, using default CRD location D:\a\dapr\dapr\charts\dapr\crds
    integration.go:86: setting up test case
    integration.go:94: starting 5 processes
    sentry.go:265: Running "sentry.exe" with args: C:\Users\RUNNER~1\AppData\Local\Temp\dapr_integration_tests\sentry.exe -log-level=info -port=6383 -issuer-ca-filename=ca.crt -issuer-certificate-filename=issuer.crt -issuer-key-filename=issuer.key -metrics-port=6385 -metrics-listen-address=127.0.0.1 -healthz-port=6384 -healthz-listen-address=127.0.0.1 -listen-address=127.0.0.1 -issuer-credentials=C:\Users\RUNNER~1\AppData\Local\Temp\Test_Integrationdaprdhotreloadoperatorinformersubscriptions1325013143\001 -trust-domain=integration.test.dapr.io -config=C:\Users\RUNNER~1\AppData\Local\Temp\Test_Integrationdaprdhotreloadoperatorinformersubscriptions1325013143\002\sentry-config.yaml
    operator.go:116: Running "operator.exe" with args: C:\Users\RUNNER~1\AppData\Local\Temp\dapr_integration_tests\operator.exe -log-level=debug -port=6392 -listen-address=127.0.0.1 -healthz-port=6394 -healthz-listen-address=127.0.0.1 -metrics-port=6393 -metrics-listen-address=127.0.0.1 -trust-anchors-file=C:\Users\RUNNER~1\AppData\Local\Temp\Test_Integrationdaprdhotreloadoperatorinformersubscriptions1325013143\007\ca.pem -disable-leader-election=true -kubeconfig=C:\Users\RUNNER~1\AppData\Local\Temp\Test_Integrationdaprdhotreloadoperatorinformersubscriptions1325013143\006\kubeconfig -webhook-server-port=6395 -webhook-server-listen-address=127.0.0.1
    daprd.go:194: Running "daprd.exe" with args: C:\Users\RUNNER~1\AppData\Local\Temp\dapr_integration_tests\daprd.exe --log-level=debug --app-id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b --app-protocol=grpc --dapr-grpc-port=6396 --dapr-http-port=6397 --dapr-internal-grpc-port=6398 --dapr-internal-grpc-listen-address=127.0.0.1 --dapr-listen-addresses=127.0.0.1 --dapr-public-port=6399 --dapr-public-listen-address=127.0.0.1 --metrics-port=6400 --metrics-listen-address=127.0.0.1 --profile-port=6401 --enable-app-health-check=false --app-health-probe-interval=0 --app-health-threshold=0 --mode=kubernetes --enable-mtls=true --enable-api-logging=true --enable-profiling --app-port=6382 --config=daprsystem --sentry-address=localhost:6383 --control-plane-address=localhost:6392 --disable-builtin-k8s-secret-store=true --control-plane-trust-domain=integration.test.dapr.io
    daprd.go:203: interrupting C:\Users\RUNNER~1\AppData\Local\Temp\dapr_integration_tests\daprd.exe process
ERROR: The process "3680" not found.
    daprd.go:203: waiting for "daprd.exe" process to exit
    daprd.go:203: "daprd.exe" process exited
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.7676257Z" level=info msg="Starting Dapr Runtime -- version edge -- commit " app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.9619445Z" level=info msg="Log level set to: debug" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.9619445Z" level=info msg="Fetching initial identity" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.security type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.9922059Z" level=info msg="Security is initialized successfully" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.security type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.9922059Z" level=info msg="Starting workload identity expiry watcher; cert expires on: 2026-03-26 19:06:07 +0000 UTC, renewal at: 2026-03-26 06:58:37 +0000 UTC" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.security type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:07.9922059Z" level=info msg="Initializing the operator client" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:08.9962433Z" level=debug msg="Loading Kubernetes config resource: daprsystem" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.0986358Z" level=info msg="Enabled features: HotReload" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.0986358Z" level=debug msg="Creating a new meter for metrics" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.0986358Z" level=info msg="metric spec: {\"enabled\":true}" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.diagnostics type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.0986358Z" level=info msg="Using default latency distribution buckets: [1 2 3 4 5 6 8 10 13 16 20 25 30 40 50 65 80 100 130 160 200 250 300 400 500 650 800 1000 2000 5000 10000 20000 50000 100000]" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.diagnostics type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.0986358Z" level=warning msg="The default value for 'spec.metric.http.increasedCardinality' will change to 'false' in Dapr 1.15 or later" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.diagnostics type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.1997367Z" level=debug msg="No resiliency policies found" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.1997367Z" level=debug msg="Found 0 resiliency configurations from Kubernetes" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.2033997Z" level=warning msg="Actor runtime disabled: api error: code = Unavailable desc = placement service is not configured. Actors and Workflow APIs will be unavailable" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.actor type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.2040311Z" level=info msg="Hot reloading enabled. Daprd will reload 'Component' and 'Subscription' resources on change." app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.hotreload type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-25T19:06:09.2040311Z" level=info msg="Actor runtime disabled: api error: code = Unavailable desc = placement service is not configured" app_id=fb9084c0-c3cb-4b00-b8c6-317f6a173b3b instance=runnervm6wf0k scope=dapr.runtime.actor type=log ver=unknown
    iowriter.go:88: Test_Integration/daprd/hotreload/operator/informer/subscriptions/daprd.exe: time="2026-03-
```

in unrelated PR build https://github.com/dapr/dapr/actions/runs/23558437587/job/68592108773?pr=9685

--- a/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go
+++ b/tests/integration/suite/daprd/hotreload/operator/informer/subscriptions.go
@@ -173,7 +173,7 @@ func (s *subscriptions) Run(t *testing.T, ctx context.Context) {
 	s.kubeapi.Informer().Modify(t, &sub2)
 	assert.EventuallyWithT(t, func(c *assert.CollectT) {
 		resp, err := s.daprd.GRPCClient(t, ctx).GetMetadata(ctx, new(rtv1.GetMetadataRequest))
-		require.NoError(t, err)
+		require.NoError(c, err)
 		if assert.Len(c, resp.GetSubscriptions(), 2) {
 			assert.Equal(c, "a", resp.GetSubscriptions()[0].GetTopic())
 			assert.Equal(c, "c", resp.GetSubscriptions()[1].GetTopic())

